### ISSUE TITLE
Generator crashes while reading metadata from IFS OData service definition

### ISSUE THREAD

**Issue by ChrisEdgington** [odata2ts/odata2ts#269]
I'm trying to convert IFS OData services to typescript types. When I run with this config:

```
import { ConfigFileOptions } from "@odata2ts/odata2ts";

const config: ConfigFileOptions = {
  services: {
    ShipmentHandling: {
      sourceUrl: "https://velvac-cfg.ifs.cloud/main/ifsapplications/projection/v1/ShipmentHandling.svc",
      source: "definitions/ShipmentHandling.xml",
      output: "generated",
    }
  }
}

export default config;
```

I get this output:

```
> npx odata2ts
Loaded config file:  odata2ts.config.ts
---------------------------
Starting generation process. Service name will be detected automatically!
Found metadata file at:  definitions/ShipmentHandling.xml
Reading metadata from file: definitions/ShipmentHandling.xml
Error while running the program TypeError: Cannot read properties of undefined (reading 'map')
    at DigesterV4.addEnum (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestion.js:208:36)
    at /Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestion.js:136:18
    at Array.forEach (<anonymous>)
    at DigesterV4.digestEntityTypesAndOperations (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestion.js:131:22)
    at DigesterV4.<anonymous> (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestion.js:123:18)
    at Generator.next (<anonymous>)
    at /Users/chris/repos/edgecraft/velvac/vsi/node_modules/tslib/tslib.js:115:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/tslib/tslib.js:111:16)
    at DigesterV4.digest (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestion.js:122:24)
    at /Users/chris/repos/edgecraft/velvac/vsi/node_modules/@odata2ts/odata2ts/lib/data-model/DataModelDigestionV4.js:13:21
    at Generator.next (<anonymous>)
    at fulfilled (/Users/chris/repos/edgecraft/velvac/vsi/node_modules/tslib/tslib.js:112:62)
```

ShipmentHandling.xml is attached. 

Versions:
```
OData 4.0.1
Node 20.12.0
```

Thanks for any help. If I get some spare time later I can try to debug if you do not have a solution.[
[ShipmentHandling.xml.txt](https://github.com/odata2ts/odata2ts/files/15266012/ShipmentHandling.xml.txt)
](url)

**Comment by texttechne** [comment]
Hi @ChrisEdgington,

this bug occurs if you have an `EnumType` without any members. Like this:

```xml
      <EnumType Name="Lookup_DecisionPowerType" IsFlags="false" UnderlyingType="Edm.Int32"></EnumType>
```

Should be an easy fix.

**Comment by texttechne** [comment]
Hi,
fixed in `@odata2ts/odata2ts` version `0.35.0`.
